src/Dialect/ONNX/ONNXOps/NN/Conv.cpp

```diff
@@ -741,138 +670,14 @@ LogicalResult ONNXConvTransposeOp::inferShapes(
-  auto spatialOffset = 2;
-  int32_t spatialRank = xShape.size() - spatialOffset;
-
-  // Use kernel_shape attribute if present otherwise use size from weight
-  // argument.
-  auto kernelShape = getKernelShape();
-  if (kernelShape.has_value()) {
-    if ((int32_t)ArrayAttrSize(kernelShape) != spatialRank) {
-      return emitError(
-          "kernel_shape length incompatible with spatial dimensions");
-    }
-    // Have the right number of values, check them.
-    for (int i = 0; i < spatialRank; ++i)
-      if (ArrayAttrIntVal(kernelShape, i) < 1) {
-        return emitError("bad kernel_shape value");
-      }
-  }
-  // Save original autoPad attribute here because it is updated in
-  // 'processConvTypeParams()'.
-  StringRef autoPad = getAutoPad();
-  // Process strides, dilations, kernel_shape and pads.
-  LogicalResult res =
-      processConvTypeParams<ONNXConvTransposeOp>(this, getX(), getW());
-  assert(succeeded(res));
-  kernelShape = getKernelShape();
-
-  auto dilationsOpt = getDilations();
-  auto stridesOpt = getStrides();
-  auto padsOpt = getPads();
-  auto outputPads = getOutputPadding();
-
-  if (autoPad == "SAME_UPPER" || autoPad == "SAME_LOWER") {
-    SmallVector<int64_t, 2> outputShapeVec;
-    for (int i = 0; i < spatialRank; ++i) {
-      outputShapeVec.emplace_back(
-          xShape[spatialOffset + i] * ArrayAttrIntVal(stridesOpt, i));
-    }
-    setOutputShapeAttr(builder.getI64ArrayAttr(llvm::ArrayRef(outputShapeVec)));
-  }
-
-  if (getOutputShape().has_value()) {
-    SmallVector<int64_t, 4> inferredPads;
-    // Determine padding values based on output shape. `pads` and
-    // `output_padding` are ignored.
-    auto outputShape = getOutputShape();
-    insertConvTransposePads(inferredPads, autoPad, xShape, kernelShape, padsOpt,
-        stridesOpt, outputPads, outputShape, dilationsOpt);
-    setPadsAttr(builder.getI64ArrayAttr(inferredPads));
-  } else {
-    llvm::SmallVector<int64_t, 4> outputShapeSpatial;
-    insertConvTransposeSpatialDim(outputShapeSpatial, xShape, kernelShape,
-        padsOpt, stridesOpt, outputPads, dilationsOpt);
-    setOutputShapeAttr(builder.getI64ArrayAttr(outputShapeSpatial));
-  }
-  auto outputShape = getOutputShape(); // spatial dimensions
-
-  llvm::SmallVector<int64_t, 4> outputShapeFinal;
-  // First two output dimensions consist of the number of batches and the
-  // number of kernels being applied.
-  // Insert batch size.
-  outputShapeFinal.emplace_back(xShape[0]);
-  // Insert number of filters being applied (number of output channels *
-  // groups).
-  outputShapeFinal.emplace_back(outChannels);
-  // Insert spatial dims.
-  for (int32_t i = 0; i < spatialRank; ++i) {
-    outputShapeFinal.emplace_back(ArrayAttrIntVal(outputShape, i));
-  }
-  getResult().setType(
-      RankedTensorType::get(outputShapeFinal, xTy.getElementType()));
-
-  return success();
+  Type elementType = getX().getType().cast<ShapedType>().getElementType();
```

Comment:
You can use `getElementType(getX().getType())` from `src/Support/TypeUtilities.hpp`.

Reply:
Done. I fixed the same pattern in the file. Thanks!